libs/auth/react/src/lib/Authenticator/CheckIdpSession.tsx

```diff
@@ -18,45 +20,42 @@ export interface SessionInfo {
   isExpired?: boolean
 }
 
-const messageEventName = 'message'
-
-const getReturnUrl = (history: History, { redirectPath }: AuthSettings) => {
-  const returnUrl = history.location.pathname + history.location.search
-  if (redirectPath && returnUrl.startsWith(redirectPath)) {
-    return '/'
-  }
-  return returnUrl
+interface SessionInfoMessage {
+  type: 'SessionInfo'
 }
 
+const messageEventName = 'message'
+
 export const CheckIdpSession = () => {
-  const history = useHistory()
   const reducerInstance = useReducer(reducer, initialState)
   const [state, dispatch] = reducerInstance
   const userManager = getUserManager()
   const authSettings = getAuthSettings()
   const iframeSrc = `${authSettings.authority}${authSettings.checkSessionPath}`
   const [iframeChecksum, setIframeChecksum] = useState(0)
+  const [sessionTimeout, setSessionTimeout] = useState<ReturnType<
+    typeof setTimeout
+  > | null>(null)
```

Comment:
Should use `useRef` for stuff like this, when we don't need/want to re-render the component.